test-suite-kotlin/src/test/kotlin/io/micronaut/http/client/SuspendClientFilter.kt

```diff
@@ -0,0 +1,35 @@
+package io.micronaut.http.client
+
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.MutableHttpRequest
+import io.micronaut.http.annotation.Filter
+import io.micronaut.http.filter.ClientFilterChain
+import io.micronaut.http.filter.HttpClientFilter
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.flatMapMerge
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.reactive.asFlow
+import kotlinx.coroutines.reactive.asPublisher
+import kotlinx.coroutines.withContext
+import org.reactivestreams.Publisher
+
+@Filter
+class SuspendClientFilter : HttpClientFilter {
+    override fun doFilter(request: MutableHttpRequest<*>, chain: ClientFilterChain): Publisher<out HttpResponse<*>> {
+        // if request contains filterCheck, then do flow step, else proceed
+        return if ((request.body.orElse(null) as? Map<*, *>)?.containsValue(filterCheck) == true) {
+            flow { emit(getValue()) }.flatMapMerge {
+                chain.proceed(request.header("myTestHeader", it)).asFlow()
```

Comment:
Are we doing anything with the HTTP Header `myTestHeader`. Should we test it?

Reply:
It is simply a usage of the value from the flow publisher. It is not necessary to test. We could easily emit and using nothing and it would have the same effect I imagine.

Reply:
ok i think it's fine as-is

Reply:
but why? if we don't verify anything. 

I would prefer to replace: 

```java
 chain.proceed(request.header("myTestHeader", it)).asFlow()
```

```java
 chain.proceed(request).asFlow()
```

thoughts?

Reply:
I see it's already merged but as long as FlowAsPublisher is somewhere in the chain it should cause the behavior.